api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -463,6 +465,25 @@ func validateAMCPVirtualNetwork(virtualNetwork ManagedControlPlaneVirtualNetwork
 	return allErrs
 }
 
+func validateFleetsMember(fleetsMember *FleetsMember, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if fleetsMember != nil {
+		match, _ := regexp.MatchString(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`, fleetsMember.Name)
```

Comment:
We perform this check because fleetsMember names can't have capital letters in Azure: https://learn.microsoft.com/en-us/rest/api/fleet/fleet-members/create?view=rest-fleet-2023-10-15&tabs=HTTP#uri-parameters

Reply:
`fleetsMember.group` is also not optional. I can make it required in this PR but I think that's a breaking API change. Should we log a warning instead if group is not specified?

Reply:
Since there are some other bugs here, I'll open a separate PR for this change

Reply:
Making this change in #4800